cli/src/test/java/com/liferay/blade/cli/command/InitCommandTest.java

```diff
@@ -592,6 +570,48 @@ private void _makeSDK(File dir) throws Exception {
 		Assert.assertTrue(Files.exists(buildCommonPluginXmlPath));
 	}
 
+	private void _testDefaultInitWorkspaceDirectory(boolean subdirectory, String workspaceName) {
+		String workspaceName1 = "workspace1";
+
+		String[] args1 = {
+			"--base", _workspaceDir.getPath(), "init", workspaceName1, "-v", BladeTest.PRODUCT_VERSION_PORTAL_74
+		};
+
+		File workspaceDir1 = new File(_workspaceDir.getPath() + File.separator + workspaceName1);
+
+		if (!workspaceDir1.exists()) {
+			TestUtil.runBlade(_workspaceDir, _extensionsDir, args1);
+		}
+
+		File baseDir = workspaceDir1;
+
+		if (subdirectory) {
+			baseDir = new File(_workspaceDir + File.separator + workspaceName1 + File.separator + "modules");
+		}
+
+		String workspaceName2 = "workspace2";
+
+		List<String> args2 = new ArrayList<>(
+			Arrays.asList("--base", baseDir.getPath(), "init", "-v", BladeTest.PRODUCT_VERSION_PORTAL_74));
+
+		if ((workspaceName != null) && !workspaceName.isEmpty()) {
+			args2.add(3, workspaceName);
+		}
+
+		BladeTestResults bladeTestResults = TestUtil.runBlade(
+			baseDir, _extensionsDir, false, args2.toArray(new String[0]));
+
+		File workspaceDir2 = new File(baseDir.getPath() + File.separator + workspaceName2);
```

Comment:
I'm a bit confused about the relationship between the `workspaceName` parameter, `workspaceName1` var, and `workspaceName2` var. Is there a way we can make this a little clearer?

Reply:
i actually struggled to find a good name for the `workspaceName` parameter.

`workspaceName1` and `workspaceName2` are the names for the different workspaces -- `workspaceName1` being the name first workspace being created, and `workspaceName2` being the optional name of the second (nested) workspace the test attempts to create.